### PR TITLE
ddl: decoupling of backfill worker and reorg worker codes, add backfill workerPool

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/37124

Problem Summary:

### What is changed and how it works?
- update some function arguments to make `backfillWorker` doesn't depend on `ddlWorker`.
- update functions position
- add backfilWorkerPool and tests
- rename function names

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/backfilling.go [hunk 1]
```diff
@@ -396,25 +374,25 @@ func (*worker) waitTaskResults(workers []*backfillWorker, taskCnt int,
 	return nextKey, addedCount, errors.Trace(firstErr)
 }
 
-// handleReorgTasks sends tasks to workers, and waits for all the running workers to return results,
+// deliveryTasksAndResults sends tasks to workers, and waits for all the running workers to return results,
 // there are taskCnt running workers.
-func (w *worker) handleReorgTasks(reorgInfo *reorgInfo, totalAddedCount *int64, workers []*backfillWorker, batchTasks []*reorgBackfillTask) error {
+func (dc *ddlCtx) deliveryTasksAndResults(sessPool *sessionPool, reorgInfo *reorgInfo, totalAddedCount *int64, workers []*backfillWorker, batchTasks []*reorgBackfillTask) error {
```
Comment (tangenta) [inline ddl/backfilling.go:379]: How about naming it with `sendTasksAndWait`?

**Review by tangenta — APPROVED** [review]
Rest LGTM

File: ddl/backfilling.go [hunk 2]
```diff
@@ -529,15 +507,15 @@ var (
 	TestCheckReorgTimeout = int32(0)
 )
 
-func loadDDLReorgVars(w *worker) error {
+func loadDDLReorgVars(ctx context.Context, sessPool *sessionPool) error {
 	// Get sessionctx from context resource pool.
-	var ctx sessionctx.Context
-	ctx, err := w.sessPool.get()
+	var sCtx sessionctx.Context
```
Comment (Defined2014) [inline ddl/backfilling.go:512]: maybe we can delete it

File: ddl/backfilling.go (hunk 1 above)
Comment (zimulala) [reply ddl/backfilling.go]: Done.

**Comment by zimulala** [comment]
PTAL @Defined2014

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
@zimulala Please fix the linter.

**Pull request merged**